flexmeasures/data/scripts/data_gen.py

```diff
@@ -254,27 +275,30 @@ def populate_time_series_forecasts(  # noqa: C901
 
     click.echo(
         "DB now has %d forecasts"
-        % db.session.query(TimedBelief)
-        .filter(TimedBelief.belief_horizon > timedelta(hours=0))
-        .count()
+        % db.session.scalar(
+            select(func.count())
+            .select_from(TimedBelief)
+            .filter(TimedBelief.belief_horizon > timedelta(hours=0))
+        )
     )
 
 
 @as_transaction
 def depopulate_structure(db: SQLAlchemy):
     click.echo("Depopulating structural data from the database %s ..." % db.engine)
-    num_assets_deleted = db.session.query(GenericAsset).delete()
-    num_asset_types_deleted = db.session.query(GenericAssetType).delete()
-    num_data_sources_deleted = db.session.query(DataSource).delete()
-    roles = db.session.query(Role).all()
+    num_assets_deleted = db.session.execute(delete(GenericAsset))
+    num_asset_types_deleted = db.session.execute(delete(GenericAssetType))
+
+    num_data_sources_deleted = db.session.execute(delete(DataSource))
+    roles = db.session.scalars(select(Role)).all()
     num_roles_deleted = 0
     for role in roles:
-        db.session.delete(role)
+        db.session.execute(delete(Role).filter_by(id=role.id))
         num_roles_deleted += 1
```

Comment:
I think this would work, too: `num_roles_deleted = db.session.execute(delete(Role))`